test/test_datasets.py

```diff
@@ -2514,6 +2514,50 @@ def _meta_to_split_and_classification_ann(self, meta, idx):
         return (image_id, class_id, species, breed_id)
 
 
+class StanfordCarsTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.StanfordCars
+    REQUIRED_PACKAGES = ("scipy",)
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(split=("train", "test"))
+
+    def inject_fake_data(self, tmpdir, config):
+        import scipy.io as io
+        from numpy.core.records import fromarrays
+
+        num_examples = {"train": 5, "test": 7}[config["split"]]
+        num_classes = 3
+        base_folder = pathlib.Path(tmpdir) / "stanford_cars"
+
+        devkit = base_folder / "devkit"
+        devkit.mkdir(parents=True)
+
+        if config["split"] == "train":
+            images_folder_name = "cars_train"
+            annotations_mat_path = str(devkit / "cars_train_annos.mat")
+        else:
+            images_folder_name = "cars_test"
+            annotations_mat_path = str(base_folder / "cars_test_annos_withlabels.mat")
+
+        datasets_utils.create_image_folder(
+            root=base_folder,
+            name=images_folder_name,
+            file_name_fn=lambda image_index: f"{image_index:5d}.jpg",
+            num_examples=num_examples,
+        )
+
+        classes = np.random.randint(1, num_classes + 1, num_examples, dtype=np.uint8)
+        fnames = [f"{i:5d}.jpg" for i in range(num_examples)]
+        rec_array = fromarrays(
+            [classes, fnames],
+            names=["class", "fname"],
+        )
+        io.savemat(annotations_mat_path, {"annotations": rec_array})
```

Comment:
Nit: I think `fromarrays` is a little overkill here. You can simply use dictionaries with `savemat` and it converts it automatically.

Reply:
saving a dict in place of `rec_array` is supported by `savemat` but I'm not sure it does what we want because when I execute the tests I get

```
  File "/Users/nicolashug/dev/vision/torchvision/datasets/stanford_cars.py", line 65, in __init__
    self._samples = [
TypeError: iteration over a 0-d array
```

So I assume our code expects a structured array here